UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Controllers/NoticestoMarinersController.cs

```diff
@@ -135,5 +138,34 @@ public IActionResult About()
         {
             return View();
         }
+
+        public async Task<ActionResult> DownloadDailyFile(string batchId, string fileName, string mimeType)
+        {
+            try
+            {
+                _logger.LogInformation(EventIds.DownloadDailyNMFileStarted.ToEventId(), "Maritime safety information request to download daily NM files started with batchId:{batchId} and fileName:{fileName} for _X-Correlation-ID:{correlationId}", batchId, fileName, GetCurrentCorrelationId());
+
+                NMHelper.ValidateParametersForDownloadSingleFile(new()
+                {
+                    new KeyValuePair<string, string>("BatchId", batchId),
+                    new KeyValuePair<string, string>("FileName", fileName),
+                    new KeyValuePair<string, string>("MimeType", mimeType)
+                }, GetCurrentCorrelationId(), _logger);
+
+                byte[] fileBytes = await _nMDataService.DownloadZipFssFile(batchId, fileName, GetCurrentCorrelationId());
+
+                _contextAccessor.HttpContext.Response.Headers.Add("Content-Disposition", "inline; filename=" + batchId + ".zip");
```

Comment:
Nicer to use interpolated string I think:
```suggestion
                _contextAccessor.HttpContext.Response.Headers.Add("Content-Disposition", $"inline; filename={batchId}.zip");
```